### PR TITLE
Ditch kiterc overrides in favor of environment variables

### LINKED ISSUE #31: Unclear how to configure .kiterc with docker
Using the docker getting started guide, I wasn't able to figure out how to configure a .kiterc file (e.g. `NEO4J_PASSWORD`).

It is referenced in the docs two places:
- https://lynxkite.com/docs/latest/user-guide/#installation
- https://lynxkite.com/docs/latest/admin-manual/#kiterc-file

Could the documentation be updated to explain how to configure these things when using Docker?

### PR DESCRIPTION
**Pull request by xandrew-lynx**
Resolves #31.

Couldn't test with docker yet. Copying stage under docker and doing a simple docker build succeeded, but then run failed with:
./lynxkite-sphynx: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by ./libnetworkit.so).

I suspect this is not due to this PR. ;)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,6 +56,7 @@ To build LynxKite you will need:
 - Java 8 and [SBT](https://www.scala-sbt.org/)
 - Python 3.7+ with [Conda](https://docs.conda.io/en/latest/miniconda.html)
 - Go 1.14+
+- Swig 4.0+
```
Comment (darabos) [inline README.md:59]: Thanks! You also need `cmake` now because it's used by NetworKit.

File: tools/call_spark_submit.sh [hunk 2]
```diff
@@ -20,30 +20,27 @@ mkdir -p ${log_dir}
...
 export SPARK_VERSION=`cat ${conf_dir}/SPARK_VERSION`
 export KITE_DEPLOYMENT_CONFIG_DIR=${conf_dir}
 export KITE_STAGE_DIR=${stage_dir}
 export KITE_LOG_DIR=${log_dir}
 if [ -f ${KITE_SITE_CONFIG} ]; then
   >&2 echo "Loading configuration from: ${KITE_SITE_CONFIG}"
+  VAR_FILE_NAME="/tmp/kite_$(randomString)_saved_env"
+  export -p > ${VAR_FILE_NAME}
   source ${KITE_SITE_CONFIG}
+  source ${VAR_FILE_NAME}
```
Comment (darabos) [inline tools/call_spark_submit.sh:36]: Sneaky! Maybe add a comment.

**Review by darabos — APPROVED** [review]
Very cool, thanks! Can you update the documentation too? I'm not sure it mentions the overrides, but I think now it should mention in the `.kiterc` section that you can override everything with plain environment variables.

> ./lynxkite-sphynx: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by ./libnetworkit.so).

Oh the horror! I got this once when I tried building with `clang` instead of `g++`. But I don't get it normally. Can you try adding `-lstdc++` to `CGO_LDFLAGS` in `sphynx/networkit/build_env.sh`?

File: README.md (hunk 1 above)
Comment (xandrew-lynx) [reply README.md:59]: Done

File: tools/call_spark_submit.sh (hunk 2 above)
Comment (xandrew-lynx) [reply tools/call_spark_submit.sh]: Done.

**Comment by darabos** [comment]
This works for me! This change is live on PizzaBox now.

File: conf/kiterc_template [hunk 3]
```diff
@@ -1,6 +1,13 @@
-
 # This file contains various configurations used when launching a LynxKite instance.
 
+# This is just a template - you need to copy to ~/.kiterc and modify as you
+# see fit. (The location ~/.kiterc can be overriden via $KITE_SITE_CONFIG.)
+
+# Environment variables set when launching LynxKite take precedence over
+# settings here! In other words, if a variable is set here as well as in
+# the launch environment than the one from the environment will be used by
+# LynxKite.
```
Comment (darabos) [inline conf/kiterc_template:9]: This note is deleted during build. This file is generated from `kiterc.asciidoc`.

**Pull request merged**